crates/symbolicator/src/services/download/mod.rs

```diff
@@ -99,12 +99,52 @@ impl DownloadService {
         source: RemoteDif,
         destination: PathBuf,
     ) -> Result<DownloadStatus, DownloadError> {
-        match source {
-            RemoteDif::Sentry(inner) => self.sentry.download_source(inner, destination).await,
-            RemoteDif::Http(inner) => self.http.download_source(inner, destination).await,
-            RemoteDif::S3(inner) => self.s3.download_source(inner, destination).await,
-            RemoteDif::Gcs(inner) => self.gcs.download_source(inner, destination).await,
-            RemoteDif::Filesystem(inner) => self.fs.download_source(inner, destination).await,
+        let result = match &source {
+            RemoteDif::Sentry(inner) => {
+                future_utils::retry(|| {
```

Comment:
Move the `retry` call up and around the `match`. I see no reason why it should be duplicated in each match arm.

Reply:
The gotcha to this is that I'm unsure how to coerce the types so that I can wrap the `match` in the `retry`. Moving it up a level produces the following error:

```
`match` arms have incompatible types
while checking the return type of the `async fn`
    expected type `impl futures::Future` (opaque type at <crates/symbolicator/src/services/download/sentry.rs:245:10>)
found opaque type `impl futures::Future` (opaque type at <crates/symbolicator/src/services/download/http.rs:71:10>)
distinct uses of `impl Trait` result in different opaque types
```

The retry expects a closure that returns a `Future`, but the futures returned by every specific `download_source` implementation are all different. In other words, if we were to move the `retry` up to wrap the `match`, every branch of that `match` is returning a different `Future` implementation. 

I could be missing a trick here to make this work. There's [future::Either](https://docs.rs/futures/0.3.15/futures/future/enum.Either.html) that might work, but it only combines two `Future`s and I'd have to chain several of them for this particular use case. Thoughts?

Something to note is that the original implementation also awaits on the future inside of every `match` branch to narrow the type down to just `Result<DownloadStatus, DownloadError>`, presumably to side-step the same issue.